contract/contracts/forward_name_resolver/forward_name_resolver.ral

```diff
@@ -0,0 +1,259 @@
+
+    @using(checkExternalCaller = false)
+    pub fn validateNFT(nftId: ByteVec, nftIndex: U256) -> () {
+        let expectedTokenContract = nftByIndex(nftIndex)
+        assert!(nftId == contractId!(expectedTokenContract), ErrorCodes.NFTNotPartOfCollection)
+    }
+
+    pub fn hasExpired(nft: Name) -> Bool {
+        let now = blockTimeStamp!()
+        let expires = nft.getExpires()
+        return expires < now
+    }
+
+    pub fn canRenew(nft: Name) -> Bool {
+        let now = blockTimeStamp!()
+        let expires = nft.getExpires()
+        return (expires - earliestRenew) < now
+    }
+
+    @using(preapprovedAssets = true, checkExternalCaller = false)
+    pub fn generateToken() -> ByteVec {
+        let tokenExists = contractExists!(subContractId!(getRewardTokenKey()))
+        assert!(!tokenExists, ErrorCodes.TokenAlreadyGenerated)
+        let caller = callerAddress!()
+
+        let (encodeImmutableFields, encodeMutableFields) = RewardToken.encodeFields!(
+            selfContractId!(),
+            b`ANS`,
+            b`Alephium Name Service`,
+            18,
+            0
+        )
+
+        return copyCreateSubContractWithToken!{caller -> ALPH: minimalContractDeposit!()}(
+            getRewardTokenKey(),
+            tokenTemplateId,
+            encodeImmutableFields,
+            encodeMutableFields,
+            u256Max!()
+        )
+    }
+
+    @using(preapprovedAssets = true)
+    fn createName(minter: Address, index: U256, name: ByteVec, capitalisation: ByteVec) -> ByteVec {
+        let now = blockTimeStamp!()
+        let expires = now + renewLength
+        let (encodeImmutableFields, encodeMutableFields) = Name.encodeFields!(
+            selfContractId!(),  // <- collectionId
+            index,              // <- nftIndex
+            name,               // <- name
+            capitalisation,     // <- capitalisation
+            minter,             // <- initialAddress
+            expires             // <- expires
+        )
+
+        let newContractId = copyCreateSubContractWithToken!{minter -> ALPH: minimalContractDeposit!()}(
+            getNftKey(index),
+            nameTemplateId,
+            encodeImmutableFields,
+            encodeMutableFields,
+            1,
+            minter
+        )
+
+        emit NameCreated(
+            index,
+            name,
+            capitalisation,
+            minter,
+            expires
+        )
+
+        return newContractId
+    }
+
+    @using(preapprovedAssets = true)
+    fn handleRewardTokens(caller: Address, name: ByteVec) -> () {
+        let nameLength = size!(name)
+        let rewardToken = getRewardToken()
+        let rewardTokenId = tokenId!(rewardToken)
+
+        if (nameLength > 8) {
+            rewardToken.mintTokens(caller, 1 alph)
+        }
+        if (nameLength <= 8) {
+            let cost = (9 - nameLength) * 100
+            rewardToken.burnTokens{caller -> rewardTokenId: cost * 1 alph}(caller, cost * 1 alph)
+        }
```

Comment:
[Nit]: Perhaps it's better to use if and else:
```
if (nameLength > 8) {
  ...
} else {
  ...
}
```

Reply:
Fixed